### PR TITLE
EVA-623 New VariantAnnotation design

### PR DESCRIPTION
**Pull request by poggio84**
- VariantAnnotation moved to project form BioData
- VariantAnnotation to DBobject converter moved to project form BioData
- Unused fields and methods removed

EVA-623

### CODE REVIEW

File: src/main/java/uk/ac/ebi/eva/commons/models/data/VariantAnnotation.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+    private String id;
+
+    private List<Xref> xrefs;
+
+    private List<String> hgvs;
+
+    private List<ConsequenceType> consequenceTypes;
+
+    private List<PopulationFrequency> populationFrequencies = null;
```
Comment (poggio84) [inline src/main/java/uk/ac/ebi/eva/commons/models/data/VariantAnnotation.java]: In _uk.ac.ebi.eva.pipeline.io.mappers.AnnotationLineMapper#parseExtraField_ populationFrequencies are populated while parsing the extra fields of VEP prediction. Is this no longer valid?

Comment (cyenyxe) [inline src/main/java/uk/ac/ebi/eva/commons/models/data/VariantAnnotation.java]: The populations it calculates frequencies for are 1000 Genomes and ExAC, which we have archived at EVA and calculated ourselves. So the ones from VEP would be redundant for us.

File: src/test/java/uk/ac/ebi/eva/commons/models/converters/data/VariantAnnotationToDBObjectConverterTest.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        Set<Integer> expectedSoAccessionsNumbers = soAccessionsValues.stream().map(
+                ConsequenceTypeMappings.termToAccession::get).collect(Collectors.toSet());
+
+        assertEquals(expectedSoAccessionsNumbers, sos);
+
+        BasicDBObject polyfhenField = (BasicDBObject) ct.get(AnnotationFieldNames.POLYPHEN_FIELD);
+        BasicDBObject siftField = (BasicDBObject) ct.get(AnnotationFieldNames.SIFT_FIELD);
+
+        assertTrue(polyphenScore.getScore().equals(polyfhenField.getDouble("sc")));
+        assertEquals(polyphenScore.getDescription(), polyfhenField.getString("desc"));
```
Comment (jmmut) [inline src/test/java/uk/ac/ebi/eva/commons/models/converters/data/VariantAnnotationToDBObjectConverterTest.java]: any reason to alternate here `assertEquals` and `assertTrue(... equals... )` ?

**Comment by cyenyxe** [comment]
Please rebase and solve the conflicts.

File: src/test/java/uk/ac/ebi/eva/pipeline/io/mappers/AnnotationLineMapperTest.java [hunk 3]
```diff
@@ -30,12 +39,96 @@
...
     }
 
+    @Test
+    public void shouldParseAllTranscriptFieldsInVepOutput() {
+        AnnotationLineMapper lineMapper = new AnnotationLineMapper();
+        VariantAnnotation variantAnnotation = lineMapper.mapLine(VepOutputContent.vepOutputContentTranscriptFields, 0);
+        List<ConsequenceType> consequenceTypes = variantAnnotation.getConsequenceTypes();
+
+        assertNotNull(consequenceTypes);
+        assertTrue(consequenceTypes.size() == 1);
```
Comment (cyenyxe) [inline src/test/java/uk/ac/ebi/eva/pipeline/io/mappers/AnnotationLineMapperTest.java]: assertEquals(1, consequenceTypes.size())

File: src/test/java/uk/ac/ebi/eva/pipeline/io/mappers/AnnotationLineMapperTest.java [hunk 4]
```diff
@@ -30,12 +39,96 @@
...
+
+        ConsequenceType consequenceType = consequenceTypes.get(0);
+
+        assertEquals(Integer.valueOf(1), consequenceType.getcDnaPosition());
+        assertEquals(Integer.valueOf(4), consequenceType.getCdsPosition());
```
Comment (cyenyxe) [inline src/test/java/uk/ac/ebi/eva/pipeline/io/mappers/AnnotationLineMapperTest.java]: assertEquals(1, consequenceType.getcDnaPosition());
assertEquals(4, consequenceType.getCdsPosition());

File: src/test/java/uk/ac/ebi/eva/pipeline/io/mappers/AnnotationLineMapperTest.java [hunk 5]
```diff
@@ -30,12 +39,96 @@
...
+    }
+
+    @Test
+    public void shouldParseVepOutputWithoutTranscript() {
+        AnnotationLineMapper lineMapper = new AnnotationLineMapper();
+        VariantAnnotation variantAnnotation = lineMapper.mapLine(VepOutputContent.vepOutputContentWithOutTranscript, 0);
+        List<ConsequenceType> consequenceTypes = variantAnnotation.getConsequenceTypes();
+
+        assertNotNull(consequenceTypes);
+        assertTrue(consequenceTypes.size() == 1);
```
Comment (cyenyxe) [inline src/test/java/uk/ac/ebi/eva/pipeline/io/mappers/AnnotationLineMapperTest.java]: assertEquals(1, consequenceTypes.size())

File: src/test/java/uk/ac/ebi/eva/pipeline/io/mappers/AnnotationLineMapperTest.java [hunk 6]
```diff
@@ -30,12 +39,96 @@
...
+
+        ConsequenceType consequenceType = consequenceTypes.get(0);
+
+        List<Score> actualScores = consequenceType.getProteinSubstitutionScores();
+        assertNotNull(actualScores);
+        assertTrue(actualScores.size() == 2);
```
Comment (cyenyxe) [inline src/test/java/uk/ac/ebi/eva/pipeline/io/mappers/AnnotationLineMapperTest.java]: assertEquals(2, actualScores.size())

File: src/test/java/uk/ac/ebi/eva/pipeline/io/mappers/AnnotationLineMapperTest.java [hunk 7]
```diff
@@ -30,12 +39,96 @@
...
+
+    @Test
+    public void shouldParseVepOutputWithExtraFields() {
+        AnnotationLineMapper lineMapper = new AnnotationLineMapper();
+        VariantAnnotation variantAnnotation = lineMapper.mapLine(VepOutputContent.vepOutputContentWithExtraFields, 0);
+
+        List<ConsequenceType> consequenceTypes = variantAnnotation.getConsequenceTypes();
+
+        assertNotNull(consequenceTypes);
+        assertTrue(consequenceTypes.size() == 1);
```
Comment (cyenyxe) [inline src/test/java/uk/ac/ebi/eva/pipeline/io/mappers/AnnotationLineMapperTest.java]: assertEquals(1, consequenceTypes.size())

File: src/main/java/uk/ac/ebi/eva/pipeline/io/mappers/AnnotationLineMapper.java [hunk 8]
```diff
@@ -196,137 +196,27 @@ private void parseExtraField(ConsequenceType consequenceType, String extraField,
...
-                case "gmaf": // Format is GMAF=G:0.2640  or  GMAF=T:0.1221,-:0.0905
-                        if(currentAnnotation.getPopulationFrequencies()==null) {
-                            currentAnnotation.setPopulationFrequencies(new ArrayList<>());
-                        }
-                        currentAnnotation.getPopulationFrequencies().add(parsePopulationFrequency(keyValue[1], "1000GENOMES",
-                                "phase_1_ALL", currentAnnotation));
-
-                    break;
                 case "hgvsc":
-                    if(currentAnnotation.getHgvs()==null) {
```
Comment (cyenyxe) [inline src/main/java/uk/ac/ebi/eva/pipeline/io/mappers/AnnotationLineMapper.java]: If this is no longer going to be checked then it must be properly initialized in the object constructor. Same with the rest of collection-based attributes, which currently are all null right after construction.

File: src/test/java/uk/ac/ebi/eva/commons/models/converters/data/DBObjectToVariantAnnotationConverterTest.java [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+                "splice_donor_variant", "SO:0001575");
+
+        assertTrue(Collections.binarySearch(consequenceTypeEntries, expectedConsequenceTypeEntry1,
+                                            consequenceTypeEntryComparator) >= 0);
+        assertTrue(Collections.binarySearch(consequenceTypeEntries, expectedConsequenceTypeEntry2,
+                                            consequenceTypeEntryComparator) >= 0);
+
+        //Xrefs
+        List<Xref> xrefList = annotation.getXrefs();
+        assertTrue(xrefList.size() == 4);
```
Comment (cyenyxe) [inline src/test/java/uk/ac/ebi/eva/commons/models/converters/data/DBObjectToVariantAnnotationConverterTest.java]: assertEquals(4, xrefList.size())

File: src/test/java/uk/ac/ebi/eva/commons/models/converters/data/VariantAnnotationToDBObjectConverterTest.java [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+        annotation.setHgvs(Arrays.asList("A", "B"));
+
+        annotation.setXrefs(Collections.singletonList(new Xref("OS01G0112100", "ensemblGene")));
+
+        DBObject dbObject = converter.convert(annotation);
+        assertNotNull(dbObject);
+
+        //Consequence types
+        LinkedList cts = (LinkedList) dbObject.get(AnnotationFieldNames.CONSEQUENCE_TYPE_FIELD);
+        assertTrue(cts.size() == 1);
```
Comment (cyenyxe) [inline src/test/java/uk/ac/ebi/eva/commons/models/converters/data/VariantAnnotationToDBObjectConverterTest.java]: assertEquals(1, cts.size())

File: src/test/java/uk/ac/ebi/eva/commons/models/converters/data/VariantAnnotationToDBObjectConverterTest.java [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+
+        VariantAnnotation annotation = new VariantAnnotation("1", 1, 2, "A", "T");
+        annotation.setConsequenceTypes(consequenceTypes);
+
+        DBObject dbObject = converter.convert(annotation);
+        assertNotNull(dbObject);
+
+        //Consequence types
+        LinkedList cts = (LinkedList) dbObject.get(AnnotationFieldNames.CONSEQUENCE_TYPE_FIELD);
+        assertTrue(cts.size() == 1);
```
Comment (cyenyxe) [inline src/test/java/uk/ac/ebi/eva/commons/models/converters/data/VariantAnnotationToDBObjectConverterTest.java]: assertEquals(1, cts.size())

File: src/test/java/uk/ac/ebi/eva/commons/models/converters/data/VariantAnnotationToDBObjectConverterTest.java [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
+
+        DBObject dbObject = converter.convert(annotation);
+
+        //Consequence types
+        assertNull(dbObject.get(AnnotationFieldNames.CONSEQUENCE_TYPE_FIELD));
+
+        //Xrefs
+        HashSet<BasicDBObject> xrefs = (HashSet) dbObject.get(AnnotationFieldNames.XREFS_FIELD);
+
+        assertTrue(xrefs.size() == 1);
```
Comment (cyenyxe) [inline src/test/java/uk/ac/ebi/eva/commons/models/converters/data/VariantAnnotationToDBObjectConverterTest.java]: assertEquals(1, xrefs.size())

**Pull request merged**